packages/react-form-state/src/FormState.ts

```diff
@@ -348,32 +371,40 @@ export default class FormState<
   }
 
   private updateRemoteErrors(errors: RemoteError[]) {
-    this.setState(({fields}: State<Fields>) => {
-      const errorDictionary = errors.reduce(
-        (accumulator: any, {field, message}) => {
-          if (field == null) {
-            return accumulator;
-          }
-
-          return set(accumulator, field, message);
-        },
-        {},
-      );
-
-      return {
-        errors,
-        fields: mapObject(fields, (field, path) => {
-          return {
-            ...field,
-            error: errorDictionary[path],
-          };
-        }),
-      };
-    });
+    this.setState(({fields, externalErrors}) => ({
+      errors,
+      fields: fieldsWithErrors(fields, [...errors, ...externalErrors]),
```

Comment:
May be simplified to `this.setState({errors});`.
See https://github.com/Shopify/quilt/pull/454/files#r245368975.